Source/Controllers/SPAppController.m

```diff
@@ -911,7 +911,20 @@ - (void)handleMySQLConnectWithURL:(NSURL *)url {
 
     NSValue *connect = @NO;
 
-    [details setObject:@"SPTCPIPConnection" forKey:@"type"];
+    if ([url query]) {
+        NSURLComponents *components = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:NO];
+        for (NSURLQueryItem *queryItem in [components queryItems]) {
+            [details setObject:queryItem.value forKey:queryItem.name];
+        }
```

Comment:
I think we need to provide some validation here! We should allow only "known" parameters to be specified and should probably present an error if someone includes an invalid parameter. This can help protect Ace from weird behavior, and can also help address user typos and other issues that may occur

Reply:
Added an NSAlert. Since I am not familiar with the error handling of ace, I decided for the most prominent option...